### PR TITLE
[CORE] Remove an extra argument of make_decimal function

### PR DESCRIPTION
**Pull request by rui-mo**
## What changes were proposed in this pull request?

Remove the extra argument `toTypeNodes` from make_decimal function.

## How was this patch tested?

Verified on Jenkins.

### CODE REVIEW

File: gluten-core/src/main/scala/io/glutenproject/expression/UnaryExpressionTransformer.scala [hunk 1]
```diff
@@ -190,12 +190,8 @@ case class MakeDecimalTransformer(
         Seq(original.dataType, BooleanType),
         FunctionConfig.OPT))
 
-    // use fake decimal literal, because velox function signature need to get return type
-    // scale and precision by input type variable
-    val toTypeNodes =
-      ExpressionBuilder.makeDecimalLiteral(new Decimal().set(0, original.precision, original.scale))
     val expressionNodes =
-      Lists.newArrayList(childNode, toTypeNodes, new BooleanLiteralNode(original.nullOnOverflow))
+      Lists.newArrayList(childNode, new BooleanLiteralNode(original.nullOnOverflow))
```
Comment (rui-mo) [inline gluten-core/src/main/scala/io/glutenproject/expression/UnaryExpressionTransformer.scala:194]: @zzcclp Is `toTypeNodes` needed for CH backend?

Comment (zzcclp) [reply gluten-core/src/main/scala/io/glutenproject/expression/UnaryExpressionTransformer.scala:194]: @loneylee please help to check this.

**Review by JkSelf — APPROVED** [review]
LGTM. Thanks.

File: gluten-core/src/main/scala/io/glutenproject/expression/UnaryExpressionTransformer.scala (hunk 1 above)
Comment (loneylee) [reply gluten-core/src/main/scala/io/glutenproject/expression/UnaryExpressionTransformer.scala:194]: Not need, but the function parameters have been changed, and the ch backend needs to update too. 

```
    else if (function_name == "make_decimal")
    {
        if (args.size() < 3)
            throw Exception(ErrorCodes::BAD_ARGUMENTS, "make_decimal function requires at least 3 args.");
        ch_function_name = SCALAR_FUNCTIONS.at(function_name);
        auto null_on_overflow = args.at(2).value().literal().boolean();
        if (null_on_overflow)
            ch_function_name = ch_function_name + "OrNull";
    }
```

CI test is passed. Maybe ut is not open.  I will commit a pr to fix it.

Comment (rui-mo) [reply gluten-core/src/main/scala/io/glutenproject/expression/UnaryExpressionTransformer.scala:194]: @loneylee Modified in this PR. Could you help review? Thanks.

**Review by JkSelf — APPROVED** [review]
Thanks.

**Comment by rui-mo** [comment]
@loneylee @JkSelf Thanks.

**Pull request merged**